src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -678,23 +705,23 @@ void CompositionalMultiphaseBase::assembleAccumulationTerms( DomainPartition con
     arrayView1d< real64 const > const & porosityRef =
       subRegion.getReference< array1d< real64 > >( viewKeyStruct::referencePorosityString() );
 
-    arrayView2d< real64 const > const & phaseVolFrac =
-      subRegion.getReference< array2d< real64 > >( viewKeyStruct::phaseVolumeFractionString() );
-    arrayView2d< real64 const > const & dPhaseVolFrac_dPres =
-      subRegion.getReference< array2d< real64 > >( viewKeyStruct::dPhaseVolumeFraction_dPressureString() );
-    arrayView3d< real64 const > const & dPhaseVolFrac_dCompDens =
-      subRegion.getReference< array3d< real64 > >( viewKeyStruct::dPhaseVolumeFraction_dGlobalCompDensityString() );
-    arrayView3d< real64 const > const & dCompFrac_dCompDens =
-      subRegion.getReference< array3d< real64 > >( viewKeyStruct::dGlobalCompFraction_dGlobalCompDensityString() );
+    arrayView2d< real64 const, compflow::USD_PHASE_PROP > const & phaseVolFrac =
+      subRegion.getReference< array2d< real64, compflow::LAYOUT_PHASE_PROP > >( viewKeyStruct::phaseVolumeFractionString() );
+    arrayView2d< real64 const, compflow::USD_PHASE_PROP > const & dPhaseVolFrac_dPres =
+      subRegion.getReference< array2d< real64, compflow::LAYOUT_PHASE_PROP > >( viewKeyStruct::dPhaseVolumeFraction_dPressureString() );
+    arrayView3d< real64 const, compflow::USD_PHASE_PROP_DC > const & dPhaseVolFrac_dCompDens =
+      subRegion.getReference< array3d< real64, compflow::LAYOUT_PHASE_PROP_DC > >( viewKeyStruct::dPhaseVolumeFraction_dGlobalCompDensityString() );
+    arrayView3d< real64 const, compflow::USD_COMP_PROP_DC > const & dCompFrac_dCompDens =
+      subRegion.getReference< array3d< real64, compflow::LAYOUT_COMP_PROP_DC > >( viewKeyStruct::dGlobalCompFraction_dGlobalCompDensityString() );
```

Comment:
Yes, precisely, the quadrature point dimension forces us to define separate layout types and USD values. As for consistency check, not really, because they can be varied independently for performance reasons, and there is currently a clear delineation between what is a constitutive data array and what isn't.

I agree that from a viewpoint of some "compute" kernel, it would be nice to just accept a slice of, say, component fractions, and not have to worry about whether it came from a constitutive model or a solver-specific field. This is unfortunately not possible with the current slice types, where a trace of the original array layout is embedded in the slice's type in the form of `USD`. Maybe if we got rid of `USD` parameter altogether... but I don't think this is going to happen.